### PR TITLE
Implement Repo selection in trigger accordion for gh PAT

### PR DESCRIPTION
**Pull request by tapishr**

### CODE REVIEW

File: components/setup/RepoSelection.tsx [hunk 1]
```diff
@@ -100,12 +100,12 @@ const RepoSelection = ({ repoProvider, installId, setIsRepoSelectionDone }: { re
...
 				} else {
 					console.info(`[RepoSelection/handleSubmit] repos data saved successfully in db`);
-					setIsRepoSelectionDone(true);
+					if(setIsRepoSelectionDone) {setIsRepoSelectionDone(true) };
 				}
 			})
 			.catch((error) => {
 				setError(`Unable to submit selected repos, \nPlease refresh this page and try again.`);
-				setIsRepoSelectionDone(false);
+				if(setIsRepoSelectionDone) {setIsRepoSelectionDone(false) };
```
Comment (MuskanPaliwal) [inline components/setup/RepoSelection.tsx:108]: can we check if the function exists before both the state setting?

**Review by MuskanPaliwal — DISMISSED** [review]
Just one comment to be addressed otherwise it seems good to me.

File: components/setup/RepoSelection.tsx (hunk 1 above)
Comment (tapishr) [reply components/setup/RepoSelection.tsx]: done

Comment (tapishr) [reply components/setup/RepoSelection.tsx]: Had to revert this. Can't return error if setIsRepoSelectionDone  is null

**Review by MuskanPaliwal — APPROVED** [review]

File: pages/docs/index.tsx [hunk 2]
```diff
@@ -96,24 +107,39 @@ const Docs = ({ bitbucket_auth_url, image_name }: { bitbucket_auth_url: string,
...
-					</AccordionContent>
+					{selectedProvider === 'github' && selectedHosting === 'selfhosting' && selectedInstallation === 'individual' ? (
+						<>
+							<AccordionTrigger>Repository Selection</AccordionTrigger>
+							<AccordionContent>
+								{installId ? (
+									<RepoSelection repoProvider={selectedProvider as RepoProvider} installId={installId as string} setIsRepoSelectionDone={null} />
+								) : (
+									<>User Info not found, please refresh and try again.</>
+								)}
```
Comment (avikalpg) [inline pages/docs/index.tsx:132]: Why did you not copy the `isGetInstallIdLoading` state as well?

Here, the user will see the failure message even when the API call to get the install id hasn't failed yet. And they might refresh the page, triggering the API again and not waiting until the API call completes.

File: components/setup/RepoSelection.tsx [hunk 3]
```diff
@@ -31,7 +31,7 @@ function formatRepoListInSaveSetupArgsForm(repos: RepoIdentifier[], install_id:
 	return Array.from(setupArgsMap.values());
 }
 
-const RepoSelection = ({ repoProvider, installId, setIsRepoSelectionDone }: { repoProvider: RepoProvider, installId: string, setIsRepoSelectionDone: Function }) => {
+const RepoSelection = ({ repoProvider, installId, setIsRepoSelectionDone }: { repoProvider: RepoProvider, installId: string, setIsRepoSelectionDone: Function | null }) => {
```
Comment (avikalpg) [inline components/setup/RepoSelection.tsx:34]: Why is this change required?

**Review by avikalpg — COMMENTED** [review]
Would like to get clarifications regarding my questions.

File: pages/docs/index.tsx (hunk 2 above)
Comment (tapishr) [reply pages/docs/index.tsx:132]: Got it, missed this for some reason. Will create an issue and a new PR for this
